src/type-exercises.ts

```diff
@@ -0,0 +1,249 @@
+ * - `C` is expected to be either `true` or `false`.
+ * - `T` and `F` can be any type.
+ *
+ * @example
+ * type A = If<true, 'a', 'b'>;  // expected to be 'a'
+ * type B = If<false, 'a', 'b'>; // expected to be 'b'
+ */
+
+// Add here your solution
+type If<C extends boolean, T, F> = C extends true ? T : F
+
+// Usage examples:
+type ClientWithoutCards = Omit<
+  Client,
+  'activeCreditCard' | 'alternativeCreditCard'
+>
+type ConditionalClient<C extends boolean> = If<C, Client, ClientWithoutCards>
+type VerifiedClient = ConditionalClient<true>
+type UnverifiedClient = ConditionalClient<false>
+
+// Add here your example
+
+/**
+ * Exercise #3: Recreate the built-in `Readonly<T>` utility type without using it.
+ *
+ * @description
+ * Constructs a type that makes all properties of `T` readonly.
+ * This means the properties of the resulting type cannot be reassigned.
+ *
+ * @example
+ * interface Todo {
+ *   title: string;
+ *   description: string;
+ * }
+ *
+ * const todo: MyReadonly<Todo> = {
+ *   title: "Hey",
+ *   description: "foobar"
+ * };
+ *
+ * todo.title = "Hello";       // Error: cannot reassign a readonly property
+ * todo.description = "barFoo"; // Error: cannot reassign a readonly property
+ */
+
+// Add here your solution
+type MyReadonly<T> = {
+  readonly [P in keyof T]: T[P]
+}
+
+type ReadOnlyClient = MyReadonly<Client>
+const readClient = client as ReadOnlyClient
+
+console.log(readClient)
+
+//Now we can't modify any property of readClient
+//readClient.active = false
+
+// Add here your example
+
+/**
+ * Exercise #4: Recreate the built-in `ReturnType<T>` utility type without using it.
+ *
+ * @description
+ * The `MyReturnType<T>` utility type extracts the return type of a function type `T`.
+ *
+ * @example
+ * const fn = (v: boolean) => {
+ *   if (v) {
+ *     return 1;
+ *   } else {
+ *     return 2;
+ *   }
+ * };
+ *
+ * type a = MyReturnType<typeof fn>; // expected to be "1 | 2"
+ */
+
+// Add here your solution
+type MyReturnType<F> = F extends (...args: any) => infer R ? R : any
+const readOrEditClient = (operation: 'read' | 'edit') => {
+  if (operation === 'edit') {
+    const verifiedClient: VerifiedClient = client as VerifiedClient
+    return verifiedClient
+  } else {
+    const unverifiedClient: UnverifiedClient = client as UnverifiedClient
+    return unverifiedClient
+  }
+}
+
+type functionType = MyReturnType<typeof readOrEditClient>
+
+// Add here your example
+
+/**
+ * Exercise #5: Extract the type inside a wrapped type like `Promise`.
+ *
+ * @description
+ * Implement a utility type `MyAwaited<T>` that retrieves the type wrapped in a `Promise` or similar structure.
+ *
+ * If `T` is `Promise<ExampleType>`, the resulting type should be `ExampleType`.
+ *
+ * @example
+ * type ExampleType = Promise<string>;
+ *
+ * type Result = MyAwaited<ExampleType>; // expected to be "string"
+ */
+
+// Add here your solution
+type MyAwaited<T extends Promise<any>> = T extends Promise<infer U> ? U : never
```

Comment:
```suggestion
type MyAwaited<T extends Promise<unknown>> = T extends Promise<infer U> ? U : never
```

Reply:
As we were discussing, unknown is a more "manageable" type at the code level compared to any. I’ll keep that in mind for future submissions. I've made the change in this commit: https://github.com/ravnhq/TS-Nederdy-Challenges/pull/52/commits/faf7bb5e960ac130c0c24f562b08cabae9d48f90